### PR TITLE
Support arbitrary path names in the TriggerResultsFilter in HLTPathStatus mode

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Support arbitrary path names in the TriggerResultsFilter in HLTPathStatus mode

Other changes:
  - clean up and modernise the module
  - add a new unit test for the HLTPathStatus mode

#### PR validation:

The new unit tests should run.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
please test

File: HLTrigger/HLTfilters/test/triggerResultsFilter_by_PathStatus.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+# empty path list
+process.filter_empty_pattern = triggerResultsFilter.clone(
+    triggerConditions = ( ),
+    l1tResults = '',
+    throw = False
+)
+
+# L1-like path name
+process.filter_l1path_pattern = triggerResultsFilter.clone(
+    triggerConditions = ( 'L1_Path', ),
```
Comment (missirol) [inline HLTrigger/HLTfilters/test/triggerResultsFilter_by_PathStatus.py:157]: Hi @fwyzard, thanks for the PR. I just wanted to ask one clarification before signing it.

In the unit test, I see that the path `L1_Path` always returns `true`, but the corresponding `TriggerResultFilter` (i.e. `filter_l1path_pattern`) always returns `false` (for `filter_l1path_pattern`, I see that `m_expression->patterns()` is empty). Is this related to how strings starting with `L1_` are handled in [`TriggerExpressionParser.h`](https://github.com/cms-sw/cmssw/blob/3707fa96313fde4b4ca7b269df402eadff1f53dd/HLTrigger/HLTcore/interface/TriggerExpressionParser.h#L26)? Could you please elaborate a bit? (I admit I don't follow the details of this)

**Comment by fwyzard** [comment]
Yes, strings of the format "L1_Something" are always matched to L1 GT/uGT
trigger results, even if a corresponding path exists in CMSSW.

If I remember correctly the Phase-2 HLT menu uses "L1T_Something" for
L1-like selections.

**Comment by missirol** [comment]
+hlt

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

**Pull request merged**